pkgs/development/compilers/ghc/8.10.4.nix

```diff
@@ -309,6 +309,14 @@ stdenv.mkDerivation (rec {
     maintainers = with lib.maintainers; [ marcweber andres peti ];
     timeout = 24 * 3600;
     inherit (ghc.meta) license platforms;
+
+    # integer-simple builds broken when compiling with musl.
+    # See https://github.com/NixOS/nixpkgs/pull/129606#issuecomment-881323743.
+    broken = enableIntegerSimple && targetPlatform.isMusl;
+    hydraPlatforms =
+      if enableIntegerSimple && targetPlatform.isMusl
+      then lib.platforms.none
+      else ghc.meta.platforms;
```

Comment:
This is almost certainly incorrect and should not be merged as-is.

Two problems with this:

1. For `broken`, I couldn't figure out what the logic should be.

    I think that GHC is broken when:
    - you're using `integer-simple`
    - you're compiling with `musl`
    - you're _not_ linking libraries statically

    On Hydra, jobs like `pkgsMusl.haskell.compiler.integer-simple.{ghc884,ghc8104,ghc901}.x86_64-linux` are failing to compile, which should match this configuration.
    
    However, I think GHC is building correctly when:
    - you're using `integer-simple`
    - you're compiling with `musl`
    - you _are_ linking libraries statically
    
    On Hydra, jobs like `pkgsStatic.haskell.packages.integer-simple.ghc8104.{hello,lens,random}.x86_64-linux` are build successfully (which should be using a GHC that matches this configuration).
    
    Is there some easy way to check whether or not libraries are being linked statically?

2.  I imagine `hydraPlatforms` is just incorrect here.  It'd probably be better just remove the things we know not to be working in `release-haskell.nix`.

Reply:
> Is there some easy way to check whether or not libraries are being linked statically?

`targetPlatform.isStatic`

> I imagine hydraPlatforms is just incorrect here. It'd probably be better just remove the things we know not to be working in release-haskell.nix.

Agreed, at the very least make `meta` into a `rec` set and do:

```
  ${if broken then "hydraPlatforms" else null} = lib.platforms.none;
```

or something like that.